astropy/utils/masked/tests/test_function_helpers.py

```diff
@@ -839,8 +839,7 @@ def test_outer(self):
         assert_masked_equal(result2, result)
 
         out2 = np.zeros_like(result.unmasked)
-        with pytest.raises(TypeError):
-            np.outer(self.ma, self.mb, out=out2)
+        np.outer(self.ma, self.mb, out=out2)
```

Comment:
Can we add a new test for the new behavior? Seems like the old behavior was to raise an exception which is no longer raised here...

Reply:
This is not generally OK: if the output does not have a mask, then it cannot hold the output of a masked operation (unless no inputs are masked). Hence, I would expect an exception to be raised.

Reply:
Moving this to a different PR/effort.